scripts/webhook-migration/updateWebhooks.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/bash
+#
+# Used to update GitHub webhooks.
+# See https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/94
+
+# Make sure we have all the required environment variables injected
+[ -z "${TENANT_URL}" ] && echo "TENANT_URL needs to be set." && exit 1
+[ -z "${AUTH_URL}" ] && echo "AUTH_URL needs to be set" && exit 1
+[ -z "${OPENSHIFT_URL}" ] && echo "OPENSHIFT_URL needs to be set." && exit 1
+[ -z "${OPENSHIFT_API_TOKEN}" ] && echo "OPENSHIFT_API_TOKEN needs to be set." && exit 1
+[ -z "${AUTH_TOKEN}" ] && echo "AUTH_TOKEN needs to be set." && exit 1
+[ -z "${JENKINS_OLD_URL_SUFFIX}" ] && echo "JENKINS_OLD_URL_SUFFIX needs to be set." && exit 1
+[ -z "${JENKINS_PROXY_URL}" ] && echo "JENKINS_PROXY_URL needs to be set." && exit 1
+[ -z "${PRIVATE_KEY}" ] && echo "PRIVATE_KEY needs to be set." && exit 1
+[ -z "${TARGET_ENV}" ] && echo "TARGET_ENV needs to be set. [prod|stage]." && exit 1
+
+DRY_RUN="${DRY_RUN:-false}"
+
+# Ensure required tools are installed
+hash jq 2>/dev/null || { echo >&2 "'jq' is required."; exit 1; }
+hash curl 2>/dev/null || { echo >&2 "'curl' is required."; exit 1; }
+hash oc 2>/dev/null || { echo >&2 "'oc' is required."; exit 1; }
+hash osio 2>/dev/null || { echo >&2 "'osio' is required."; exit 1; }
+
+###############################################################################
+# Processes the build configuration of
+# the OpenShift cluster and extracts GitHub
+# repo URLs which might need updating.
+# Globals:
+#   OPENSHIFT_URL
+#   OPENSHIFT_API_TOKEN
+#   NAME_SPACES
+#   REPOS
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+findGitHubRepos() {
+  export KUBECONFIG=/tmp/config
+  echo "Using KUBECONFIG ${KUBECONFIG}"
+  oc login ${OPENSHIFT_URL} --token=${OPENSHIFT_API_TOKEN} > /dev/null
+
+  local projects=$(oc projects -q)
+  for project in ${projects}; do
+   echo "Processing project ${project}"
+   oc project ${project} > /dev/null;
+   local tmp=($(oc get bc -o json | jq -r ' .items[] | select(.spec.source.git.uri) | .metadata.namespace + " " + .spec.source.git.uri[19:-4]'))
```

Comment:
Also just realised - isn't `namespace == project` always?

Reply:
Yes, I guess we could also use _namepace_ as the local variable name. Might be less confusing